### PR TITLE
Auto detect file tracking method

### PR DESCRIPTION
**Pull request by TrevCraw**
Fixes https://github.com/OpenLiberty/ci.maven/issues/748

### CODE REVIEW

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java [hunk 1]
```diff
@@ -1288,6 +1304,16 @@ public void watchFiles(File buildFile, File outputDirectory, File testOutputDire
                     Thread.sleep(pollingInterval);
                 }
             }
+        } finally {
+            if (watcher != null) {
+                try {
+                    watcher.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
```
Comment (ericglau) [inline src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java:1313]: Log the exception using `error(.....)`

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java [hunk 2]
```diff
@@ -1351,50 +1378,74 @@ public boolean accept(File file) {
...
+                            if (watcher != null) {
+                                watcher.close();
+                            }
+                        } else {
+                            trackingMode = FileTrackMode.FILE_WATCHER;
+                            disablePolling();
+                        }
+                    } catch (Exception e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
```
Comment (ericglau) [inline src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java:1444]: Log the exception using `error(.....)`

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java (hunk 1 above)
Comment (TrevCraw) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java]: How is this?
```
catch (IOException e) {
                    error("An error occurred attempting to close the file watcher. " + e.getMessage());
                }
```

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java (hunk 2 above)
Comment (TrevCraw) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java]: How is this?
```
catch (Exception e) {
                        error("An error occured attempting to retrieve the watch key or close the file watcher. " + e.getMessage());
                    }
```

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java (hunk 1 above)
Comment (ericglau) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java]: Also include the throwable as the second parameter so the stacktrace can be logged. e.g. use `error(String msg, Throwable e)`

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java (hunk 2 above)
Comment (ericglau) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java]: Also include the throwable as the second parameter so the stacktrace can be logged. e.g. use `error(String msg, Throwable e)`

**Pull request merged**